### PR TITLE
Mp 1170

### PR DESCRIPTION
**Pull request by weteringac**

### CODE REVIEW

File: ada_2_ada/ada/AddConceptIds.xsl [hunk 1]
```diff
@@ -50,7 +54,10 @@
     <xsl:template match="/">
         <xsl:choose>
             <xsl:when test="empty($schema)">
-                <xsl:message>Schema empty, please check</xsl:message>
+                <xsl:call-template name="util:logMessage">
+                    <xsl:with-param name="level" select="$logERROR"/>
```
Comment (LuudSlagter) [inline ada_2_ada/ada/AddConceptIds.xsl:58]: This seems like a similar message as the one above, should both be either WARN or ERROR?

File: ada_2_fhir-r4/fhir/2_fhir_fhir_include.xsl [hunk 2]
```diff
@@ -496,7 +520,8 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
                             <xsl:with-param name="in" select="($identifier[not(@root = $mask-ids-var)], $identifier)[1]"/>
                         </xsl:call-template>
                     </identifier>
-                </xsl:when> <!--<xsl:when test="local-name() = $adaElement and .[@value]">
+                </xsl:when>
+                <!--<xsl:when test="local-name() = $adaElement and .[@value]">
```
Comment (LuudSlagter) [inline ada_2_fhir-r4/fhir/2_fhir_fhir_include.xsl:524]: Should we just remove this commented-out code?

File: fhir_2_ada-r4/mp/9.2.0/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl [hunk 3]
```diff
@@ -40,7 +40,10 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
 
 					<xsl:choose>
 						<xsl:when test="count(f:Bundle/f:entry/f:resource/f:Patient) ge 2 or count(distinct-values(f:Bundle/f:entry/f:resource/(f:MedicationRequest | f:MedicationDispense | f:MedicationStatement)/f:subject/f:reference/@value)) ge 2">
-							<xsl:message terminate="yes">Multiple Patients and/or subject references found. Please check.</xsl:message>
+							<xsl:call-template name="util:logMessage">
+								<xsl:with-param name="level" select="$logWARN"/>
```
Comment (LuudSlagter) [inline fhir_2_ada-r4/mp/9.2.0/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl:44]: Perhaps these types of messages should be ERROR; see similar messages in e.g. contraindicatie and lab. In the files below this should be changed as well (if we choose to do so).

File: fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl [hunk 4]
```diff
@@ -26,31 +26,31 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
...
     <xsl:import href="ext-TimeInterval-Duration.xsl"/>
+    <xsl:import href="ext-TimeInterval-period.xsl"/>
     <xsl:import href="nl-core-AddressInformation.xsl"/>
     <xsl:import href="nl-core-AdministrationAgreement.xsl"/>
     <xsl:import href="nl-core-BodyHeight.xsl"/>
     <xsl:import href="nl-core-BodyWeight.xsl"/>
     <xsl:import href="nl-core-ContactInformation.xsl"/>
     <xsl:import href="nl-core-ContactPerson.xsl"/>
+    <xsl:import href="nl-core-DispenseRequest.xsl"/>
+    <xsl:import href="nl-core-ext-StopType.xsl"/>
```
Comment (LuudSlagter) [inline fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl:38]: Why not just ext-StopType (see extensions above)?

File: fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl [hunk 5]
```diff
@@ -26,31 +26,31 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
     <xsl:import href="contextContactEpisodeOfCare.xsl"/>
     <xsl:import href="ext-CopyIndicator.xsl"/>
     <xsl:import href="ext-MedicationAgreementPeriodOfUseCondition.xsl"/>
-    <xsl:import href="ext-TimeInterval-period.xsl"/>
     <xsl:import href="ext-TimeInterval-Duration.xsl"/>
+    <xsl:import href="ext-TimeInterval-period.xsl"/>
```
Comment (LuudSlagter) [inline fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl:30]: ext-TimeInterval-Period (?)
Or even better: ext-TimeInterval.Period (in line with the name of the extension)

File: fhir_2_ada-r4/zibs2020/payload/0.8.0-beta.1/all_zibs.xsl [hunk 6]
```diff
@@ -26,27 +26,31 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
     <xsl:import href="contextContactEpisodeOfCare.xsl"/>
     <xsl:import href="ext-CopyIndicator.xsl"/>
     <xsl:import href="ext-MedicationAgreementPeriodOfUseCondition.xsl"/>
-    <xsl:import href="ext-TimeInterval-period.xsl"/>
     <xsl:import href="ext-TimeInterval-Duration.xsl"/>
+    <xsl:import href="ext-TimeInterval-period.xsl"/>
```
Comment (LuudSlagter) [inline fhir_2_ada-r4/zibs2020/payload/0.8.0-beta.1/all_zibs.xsl:30]: See remarks in 0.0 version.

File: fhir_2_ada/mp/9.0.7/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl [hunk 7]
```diff
@@ -46,7 +47,10 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
                     
                     <xsl:choose>
                         <xsl:when test="count(f:Bundle/f:entry/f:resource/f:Patient) ge 2 or count(distinct-values(f:Bundle/f:entry/f:resource/(f:MedicationRequest|f:MedicationDispense|f:MedicationStatement)/f:subject/f:reference/@value)) ge 2">
-                            <xsl:message terminate="yes">Multiple Patients and/or subject references found. Please check.</xsl:message>
+                            <xsl:call-template name="util:logMessage">
+                                <xsl:with-param name="level" select="$logWARN"/>
```
Comment (LuudSlagter) [inline fhir_2_ada/mp/9.0.7/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl:51]: Consider to make this an ERROR (also files below).

File: hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl [hunk 8]
```diff
@@ -418,7 +424,10 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
                     </xsl:call-template>
                 </xsl:when>
                 <xsl:otherwise>
-                    <xsl:message terminate="yes">Cannot determine the datatype based on @xsi:type, or value not supported: <xsl:value-of select="$xsiType"/>. Calculated QName <xsl:value-of select="$xsiTypeURIName"/></xsl:message>
+                    <xsl:call-template name="util:logMessage">
+                        <xsl:with-param name="level" select="$logWARN"/>
```
Comment (LuudSlagter) [inline hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl:428]: This should be FATAL based on `terminate="yes"` in old situation.

File: hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl [hunk 9]
```diff
@@ -1435,7 +1447,11 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
         </xsl:if>
         <xsl:for-each select="$in">
             <xsl:if test="@value[not(. = ('true', 'false'))]">
-                <xsl:message terminate="yes">FATAL: Message contains illegal boolean value. Expected 'true' or 'false'. Found: "<xsl:value-of select="$in/@value"/>" </xsl:message>
+                <xsl:call-template name="util:logMessage">
+                    <xsl:with-param name="level" select="$logFATAL"/>
+                    <xsl:with-param name="msg">FATAL: Message contains illegal boolean value. Expected 'true' or 'false'. Found: "<xsl:value-of select="$in/@value"/>" </xsl:with-param>
```
Comment (LuudSlagter) [inline hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl:1452]: Remove ' FATAL: '  from `msg`.

File: hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl [hunk 10]
```diff
@@ -66,7 +66,11 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
     <xsl:template match="hl7:QUPC_IN990002NL">
         <xsl:choose>
             <xsl:when test="ancestor::hl7:MCCI_IN200101">
-                <xsl:message terminate="yes">Unsupported input MCCI_IN200101. Can only support singular QUPC_IN990002NL</xsl:message>
+                <xsl:call-template name="util:logMessage">
+                    <xsl:with-param name="level" select="$logWARN"/>
```
Comment (LuudSlagter) [inline hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl:70]: This should be FATAL based on old version.

File: hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl [hunk 11]
```diff
@@ -1395,7 +1399,10 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
                             <valueBoolean value="{hl7:value/@value}"/>
                         </xsl:when>
                         <xsl:when test="hl7:value">
-                            <xsl:message>Niet-ondersteund datatype <xsl:value-of select="@xsi:type"/></xsl:message>
+                            <xsl:call-template name="util:logMessage">
+                                <xsl:with-param name="level" select="$logWARN"/>
+                                <xsl:with-param name="msg">Niet-ondersteund datatype <xsl:value-of select="@xsi:type"/></xsl:with-param>
```
Comment (LuudSlagter) [inline hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl:1404]: Translate `msg` to English, as above `msg` is also in English (?)

**Review by LuudSlagter — CHANGES_REQUESTED** [review]

File: ada_2_fhir-r4/cio/2.0.0/beschikbaarstellen_overgevoeligheid/payload/beschikbaarstellen_overgevoeligheid_2_fhir.xsl [hunk 12]
```diff
@@ -974,10 +975,17 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
...
+                <xsl:call-template name="util:logMessage">
+                    <xsl:with-param name="level" select="$logFATAL"/>
+                    <xsl:with-param name="msg">_insertFullUrlById: Nothing found. (<xsl:value-of select="count($fhirId)"/>)</xsl:with-param>
+                    <xsl:with-param name="terminate" select="true()"/>
+                </xsl:call-template>
             </xsl:when>
             <xsl:when test="count($theMetaData) gt 1">
-                <xsl:message terminate="no">_insertFullUrlById: Multiple found (<xsl:value-of select="count($theMetaData)"/>): <xsl:value-of select="$fhirId"/> - <xsl:value-of select="$theMetaData/nm:full-url"/></xsl:message>
+                <xsl:call-template name="util:logMessage">
+                    <xsl:with-param name="level" select="$logWARN"/>
```
[the remark this answers is not in the document]
Comment (weteringac) [reply ada_2_fhir-r4/cio/2.0.0/beschikbaarstellen_overgevoeligheid/payload/beschikbaarstellen_overgevoeligheid_2_fhir.xsl:986]: Should be error unless someone explicitly stated warning. Will update to error.

File: ada_2_ada/ada/AddConceptIds.xsl (hunk 1 above)
Comment (weteringac) [reply ada_2_ada/ada/AddConceptIds.xsl:58]: No this error is when there is no schema whatsoever, that is more serious than when an individual element cannot be found ín an existing schema.

File: ada_2_fhir-r4/fhir/2_fhir_fhir_include.xsl (hunk 2 above)
Comment (weteringac) [reply ada_2_fhir-r4/fhir/2_fhir_fhir_include.xsl:524]: Agree!

File: fhir_2_ada-r4/mp/9.2.0/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl (hunk 3 above)
Comment (weteringac) [reply fhir_2_ada-r4/mp/9.2.0/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl:44]: Agree

File: fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl (hunk 4 above)
Comment (weteringac) [reply fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl:38]: Good question but out of scope for this ticket. Created new ticket: MP-1189

File: fhir_2_ada/mp/9.0.7/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl (hunk 7 above)
Comment (weteringac) [reply fhir_2_ada/mp/9.0.7/beschikbaarstellen_medicatiegegevens/payload/beschikbaarstellen_medicatiegegevens_2_ada.xsl:51]: Agree!

File: hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl (hunk 8 above)
Comment (weteringac) [reply hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl:428]: Agree!

File: hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl (hunk 9 above)
Comment (weteringac) [reply hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl:1452]: Thanks

File: hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl (hunk 10 above)
Comment (weteringac) [reply hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl:70]: Agree

File: hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl (hunk 11 above)
Comment (weteringac) [reply hl7_2_fhir/huisartswaarneemgegevens/6.10.1.3/professionele_samenvatting/payload/QUPC_IN990002NL-2-BundleGeneralPractitioner.xsl:1404]: Agree, formally out of scope for this ticket, but updated it

File: hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl (hunk 8 above)
Comment (weteringac) [reply hl7_2_ada/hl7/hl7_2_ada_hl7_include.xsl]: Agree

File: fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl (hunk 5 above)
Comment (weteringac) [reply fhir_2_ada-r4/zibs2020/payload/0.0/all_zibs.xsl:30]: Good question but out of scope for this ticket. Created new ticket: MP-1189

File: fhir_2_ada-r4/zibs2020/payload/0.8.0-beta.1/all_zibs.xsl (hunk 6 above)
Comment (weteringac) [reply fhir_2_ada-r4/zibs2020/payload/0.8.0-beta.1/all_zibs.xsl:30]: Good question but out of scope for this ticket. Created new ticket: MP-1189

**Review by LuudSlagter — APPROVED** [review]

**Review by ogieling — APPROVED** [review]

**Pull request merged**